storagemanager/storage/views.py

```diff
@@ -47,33 +47,68 @@ def delete_item(request):
     return render(request, 'storage/delete_item.html', context)
 
 
-def delete_item_confirm(request):
-    if request.POST.get('id') == '':
-        messages.warning(request, f'Please type a number')
-        return redirect('delete-item')
-    elif 'e' in request.POST.get('id'):
-        messages.warning(request, f'Exponent numbers are not allowed')
-        return redirect('delete-item')
+def __was_item_chosen_from_the_list(chosen_item):
+    if chosen_item.startswith('list_item_'):
```

Comment:
This is very hacky solution - adding it to the chosen value and then manually stripping it from it. Maybe try to debug `request` object that is being provided to the view and see if there is any data in it that could tell you from which input current value was provided? Or maybe you can add `id` to these html elements and then see them somewhere in `request`?
To debug, you can simply `print(request.__dict__)` to see its internals or try to use `import import pdb; pdb.set_trace()` by putting it in view, run your app and this view. Then your app should break in this place and in terminal you will be able to write i.e. `request` and see what's in it.

Reply:
After debugging (nice tool btw, thanks :D ), I couldn't find anything specific in request.POST beside the CSRF token, but I don't think that's the way to go.
But here is an idea - I could add names to the buttons (since I have separate button for every form), and add that name to if statement, to valid which form was used. What do you think? I was trying to do that with form's name, but it doesn't seem to be sent in request.POST data.

Reply:
Yeah, sound much much better than current hack. Go for it :rocket: